core/src/main/java/io/confluent/rest/auth/AuthUtil.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest.auth;
+
+import io.confluent.rest.RestConfig;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.eclipse.jetty.security.ConstraintMapping;
+import org.eclipse.jetty.util.security.Constraint;
+
+public final class AuthUtil {
+
+  private AuthUtil() {
+  }
+
+  public static boolean isCorsEnabled(final RestConfig restConfig) {
+    String allowedOrigins = restConfig.getString(RestConfig.ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG);
+    return !allowedOrigins.trim().isEmpty();
+  }
+
+  /**
+   * Build the standard global auth constraint from standard RestConfig.
+   *
+   * <p>The valid roles is extracted from {@link RestConfig#AUTHENTICATION_ROLES_CONFIG}
+   *
+   * <p>OPTIONS requests will not require auth if
+   * {@link RestConfig#ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG} is not empty.
+   *
+   * @param restConfig the rest app's config.
+   * @return the constraint mapping
+   */
+  public static ConstraintMapping createGlobalAuthConstraint(final RestConfig restConfig) {
+    final List<String> roles = restConfig.getList(RestConfig.AUTHENTICATION_ROLES_CONFIG);
+    final boolean omitOptions = isCorsEnabled(restConfig);
+    return createGlobalAuthConstraint(roles, omitOptions);
+  }
+
+  /**
+   * Build the standard global auth constraint.
+   *
+   * <p>To be authorised a user must below to at least one of the supplied {@code roles}.
+   *
+   * <p>If {@code omitOptions} is {@code true} the HTTP OPTIONS requests are excluded from
+   * auth checks, e.g. as required for CORS preflight checks.
+   *
+   * @param roles the list of valid roles
+   * @param omitOptions if {@code true}, OPTIONS requests are not subject to auth checks.
+   * @return the constraint mapping.
+   */
+  public static ConstraintMapping createGlobalAuthConstraint(
```

Comment:
let's remove `Application.createGlobalAuthConstraint(List<String> roles)` so we can simplify things and merge this into a single `createGlobalAuthConstraint(final RestConfig restConfig)`

Reply:
Done